rustic-cargo.el

```diff
@@ -459,19 +467,68 @@ If running with prefix command `C-u', read whole command from minibuffer."
                              'compile-history)))))
     (rustic-run-cargo-command command (list :mode 'rustic-cargo-run-mode))))
 
-(define-derived-mode rustic-cargo-run-mode rustic-compilation-mode "Cargo run"
-  "Mode for 'cargo run' that derives from `rustic-compilation-mode', but uses
-the keymap of `comint-mode' so user input is possible."
+(defun rustic-cargo-run-mode ()
+  (interactive)
+  (if rustic-cargo-run-use-comint
+      ;; rustic-cargo-comint-run-mode toggles the mode; we want to
+      ;; always enable.
+      (unless (and (boundp 'polymode-mode)
+                   polymode-mode
+                   (memq major-mode '(rustic-cargo-plain-run-mode
+                                      comint-mode)))
+        (rustic-cargo-comint-run-mode))
+    (rustic-cargo-plain-run-mode)))
+
+(define-derived-mode rustic-cargo-plain-run-mode rustic-compilation-mode "Cargo run"
+  "Mode for 'cargo run' that derives from `rustic-compilation-mode'.
+
+To send input to the compiled program, use
+`rustic-compile-send-input'.  If you set
+`rustic-cargo-run-use-comint' to t, you can also just type in a
+string and hit RET to send it to the program.  The latter
+approach requires installing polymode."
   (buffer-disable-undo)
   (setq buffer-read-only nil)
-  (use-local-map comint-mode-map)
-
-  (define-key (current-local-map) (kbd "RET")
-    '(lambda ()
-       (interactive)
-       (when (eq major-mode 'rustic-cargo-run-mode)
-         (process-send-string
-          (get-buffer-process (current-buffer)) "\n")))))
```

Comment:
We didn't merge this code to master, which branch did you checkout ?

Reply:
This is based on the cargo-run-send-input branch.

Reply:
Can you change the target branch to master or is there a reason you didn't use master ?